Riot/Utils/EventFormatter.m

```diff
@@ -425,7 +425,9 @@ - (NSString*)senderAvatarUrlForEvent:(MXEvent*)event withRoomState:(MXRoomState*
     
     // Check whether this avatar url is updated by the current event (This happens in case of new joined member)
     NSString* membership = event.content[@"membership"];
-    if (membership && [membership isEqualToString:@"join"] && [event.content[@"avatar_url"] length])
+    NSString* eventAvatarUrl = event.content[@"avatar_url"];
+    NSString* prevEventAvatarUrl = event.prevContent[@"avatar_url"];
+    if (membership && [membership isEqualToString:@"join"] && [eventAvatarUrl length] && ![eventAvatarUrl isEqualToString:prevEventAvatarUrl])
     {
         // Use the actual avatar
         senderAvatarUrl = event.content[@"avatar_url"];
```

Comment:
I imagine this should use `eventAvatarUrl` instead of going to the dictionary again.